express-app/controllers/user.js

```diff
@@ -109,4 +127,21 @@ module.exports = {
       console.log(error);
     });
   },
+
+  /**
+  * This function reset the password from a user that was
+  * actually logged.
+  *
+  * @function
+  * @param {object} user - actual user who will be edited.
+  * @param {object} data - user updated
+  */
+  resetPassword: function (user, data) {
+    let hash = bcrypt.hashSync(data.password, bcrypt.genSaltSync(10), null);
+    User.findOneAndUpdate({ _id: user._id }, { password: hash }).then((editedUser) => {
```

Comment:
yes, no it hash only one time  in preSave() is only when is save(). When is updated it doesn't enter in the function.